game/core/src/me/lihq/game/screen/NavigationScreen.java

```diff
@@ -229,11 +235,17 @@ public void finishRoomTransition()
     @Override
     public void render(float delta)
     {
-
         game.player.pushCoordinatesToSprite();
+        for (AbstractPerson n : currentNPCS) {
+            n.pushCoordinatesToSprite();
+
+        }
 
         if (changeMap) {
             tiledMapRenderer.setMap(game.player.getRoom().getTiledMap());
+            tiledMapRenderer.clearPeople();
+            tiledMapRenderer.addPerson((List<AbstractPerson>) ((List<? extends AbstractPerson>) currentNPCS));
```

Comment:
I originally tried what you said, which works, however then when you call getNPCs your not getting the NPC's, I thought having one line that was a bit messy was better than having the whole thing doing something different